src/main/java/com/github/princesslana/eriscasper/action/ActionContext.java

```diff
@@ -0,0 +1,14 @@
+package com.github.princesslana.eriscasper.action;
+
+import com.github.princesslana.eriscasper.data.util.Nullable;
+import com.github.princesslana.eriscasper.gateway.Gateway;
+import com.github.princesslana.eriscasper.rest.Routes;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface ActionContext {
+
+  Nullable<Routes> getRoutes();
```

Comment:
In what cases do we expect this to be null?

Reply:
Alright, I may keep the gateway null since it makes sense that you can provide routes without a gateway to process commands through. I'll need to redesign some of the tests we have though, due to the removal of the `Nullable<Routes>`. May take a bit of work just since in test cases for certain bot context methods we aren't able to produce a viable `Routes` but then again we can make a FakeRoutes class. We could also make a FakeGateway pass through as well, if we want to remove the nullable attribute from that as well (which might be a good idea)

Reply:
You can easily make a mock Routes or Gateway for testing.

The only place we create an `ActionContext`, is from `BotContext`. There are no cases we have at the moment that would require us to create an `ActionContext` without a `Gateway`.

Reply:
Would just using Mockito provide enough mocking for the Routes and Gateway?

Reply:
For testing, yes.